ppl/src/test/java/org/opensearch/sql/ppl/parser/AstBuilderTest.java

```diff
@@ -93,6 +95,29 @@ public void testSearchCommandWithDotInIndexName() {
     );
   }
 
+  @Test
+  public void testSearchWithPrometheusQueryRangeWithPositionedArguments() {
+    assertEqual("search source = prometheus.query_range(\"test{code='200'}\",1234, 12345, 3)",
+        tableFunction("prometheus.query_range",
+            unresolvedArg(null, stringLiteral("test{code='200'}")),
+            unresolvedArg(null, intLiteral(1234)),
+            unresolvedArg(null, intLiteral(12345)),
+            unresolvedArg(null, intLiteral(3))
+    ));
+  }
+
+  @Test
+  public void testSearchWithPrometheusQueryRangeWithNamedArguments() {
+    assertEqual("search source = prometheus.query_range(query = \"test{code='200'}\", "
+            + "starttime = 1234, step=3, endtime=12345)",
+        tableFunction("prometheus.query_range",
+            unresolvedArg("query", stringLiteral("test{code='200'}")),
+            unresolvedArg("starttime", intLiteral(1234)),
+            unresolvedArg("step", intLiteral(3)),
+            unresolvedArg("endtime", intLiteral(12345))
```

Comment:
Try to reorder arguments to ensure that they are resolved by name.

Reply:
Ast Builder doesn't care about the order and pass the same to Analyzer where the logic of analyzing arguments happen.  I have included these test cases in Analyzer Test class.